LayoutKit/Layouts/ButtonLayout.swift

```diff
@@ -0,0 +1,223 @@
+// Copyright 2016 LinkedIn Corp.
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+import UIKit
+
+/**
+ Layout for a UIButton.
+
+ Since UIKit does not provide threadsafe methods to determine the size of a button given its content
+ ButtonLayout's implementation is not future proof for any design changes that may happen to UIButton in the future.
+ If the style of UIButton changes in the future, then ButtonLayout will need to be updated accordingly.
+
+ If future-proofing is a concern for your application, then you should not use ButtonLayout and instead implement your own
+ custom layout that uses you own custom button view (e.g. by subclassing UIControl).
+ */
+open class ButtonLayout<Button: UIButton>: BaseLayout<Button>, ConfigurableLayout {
+
+    private let type: ButtonLayoutType
```

Comment:
Why make these private? I passed them in the initializer, so seems like being able to read them may be helpful?